packages/preferences/src/browser/preferences-tree-widget.ts

```diff
@@ -245,6 +250,17 @@ export class PreferencesContainer extends SplitPanel implements ApplicationShell
             await this.editorsContainer.refreshFoldersPreferencesEditorWidget(firstRoot ? firstRoot.uri : undefined);
         }
     }
+
+    /**
+     * Show a message when there are unsaved preference editors.
+     * Users will have the opportunity to save and retry an edit.
+     * @returns a Promise that resolves to the user's selection or `undefined`.
+     */
+    private showDirtyMessage(): Promise<string | undefined> {
+        const MESSAGE: string = 'Preferences editor(s) has/have unsaved changes';
+        const SAVE_RETRY: string = 'Save and Retry';
+        return Promise.resolve(this.messageService.warn(MESSAGE, SAVE_RETRY));
```

Comment:
Do we need the variables for the text? It is not even reused elsewhere, we should be able to do without.

Also, `this.messageService.warn()` already returns a promise, please remove `Promise.resolve`.